code/_helpers/game.dm

```diff
@@ -128,32 +128,31 @@
 // Will recursively loop through an atom's contents and check for mobs, then it will loop through every atom in that atom's contents.
 // It will keep doing this until it checks every content possible. This will fix any problems with mobs, that are inside objects,
 // being unable to hear people due to being in a box within a bag.
+// Does not return list, as list is passed as reference.
 
 /proc/recursive_content_check(var/atom/O,  var/list/L = list(), var/recursion_limit = 3, var/client_check = 1, var/sight_check = 1, var/include_mobs = 1, var/include_objects = 1)
 
 	if(!recursion_limit)
-		return L
+		return 
 
 	for(var/I in O.contents)
```

Comment:
While we're optimizing things, you could change this block to be like:

```suggestion
	for(var/mob/M in O.contents)
	    ...
	for(var/obj/I in O.contents)
	    ...
```

Byond has some optimization for its built-in types like /mob, /obj and from my tests just now it is significantly faster even if we lose the 'else if' and check things twice.

Reply:
I didn't know that, thanks for bringing it up

Reply:
When testing that in Lohikar's DM bot the benchmarks show. In the list where there are 2 times more obj than mobs gives:
```
looping using for(var/i in O): 20 ds
looping over objects for(var/obj/o in O): 31 ds
looping over mobs for(var/mob/m in O): 11 ds
```

Also, the contents of this will only have either mob or object types, that's why we won't gain anything. As you can see if you combine together time for looping over obj and mobs that's 41ds and looping over all of them without type casting is 20s, which is 2x faster.

Reply:
Interesting, I wonder why I had the opposite result in testing. How many things were in your lists?

Reply:
I've heard of this before. This optimization may only apply for special lists. So test it exclusively on the contents list, as that's what's relevant here.

Reply:
Seems like there were 3000 items in the list. 2000 obj and 1000 mobs.